lib/tools/broccoli/node-modules-dest-copy.ts

```diff
@@ -30,41 +30,43 @@ export class DestCopy implements IBroccoliPlugin {
   
   public rebuildChangedDirectories(changedDirectories: string[], platform: string): void {
 	_.each(changedDirectories, changedDirectoryAbsolutePath => {
-		let pathToPackageJson = path.join(changedDirectoryAbsolutePath, "package.json");
-		let packageJsonFiles = fs.existsSync(pathToPackageJson) ? [pathToPackageJson] : [];
-		let nodeModulesFolderPath = path.join(changedDirectoryAbsolutePath, "node_modules");
-		packageJsonFiles = packageJsonFiles.concat(this.enumeratePackageJsonFilesSync(nodeModulesFolderPath));
-			
-		_.each(packageJsonFiles, packageJsonFilePath => {
-			let fileContent = require(packageJsonFilePath);
-			let isPlugin = fileContent.nativescript;
-	
-			if(!this.devDependencies[fileContent.name]) { // Don't flatten dev dependencies
-		
-				let currentDependency = {
-					name: fileContent.name,
-					version: fileContent.version,
-					directory: path.dirname(packageJsonFilePath),
-					isPlugin: isPlugin
-				};
+		if(!this.devDependencies[path.basename(changedDirectoryAbsolutePath)]) {
+			let pathToPackageJson = path.join(changedDirectoryAbsolutePath, "package.json");
+			let packageJsonFiles = fs.existsSync(pathToPackageJson) ? [pathToPackageJson] : [];
+			let nodeModulesFolderPath = path.join(changedDirectoryAbsolutePath, "node_modules");
+			packageJsonFiles = packageJsonFiles.concat(this.enumeratePackageJsonFilesSync(nodeModulesFolderPath));
 				
-				let addedDependency = this.dependencies[currentDependency.name];
-				if (addedDependency) {
-					if (semver.gt(currentDependency.version, addedDependency.version)) {
-						let currentDependencyMajorVersion = semver.major(currentDependency.version);
-						let addedDependencyMajorVersion = semver.major(addedDependency.version);
-		
-						let message = `The depedency located at ${addedDependency.directory} with version  ${addedDependency.version} will be replaced with dependency located at ${currentDependency.directory} with version ${currentDependency.version}`;
-						let logger = $injector.resolve("$logger");
-						currentDependencyMajorVersion === addedDependencyMajorVersion ? logger.out(message) : logger.warn(message);
+			_.each(packageJsonFiles, packageJsonFilePath => {
+				let fileContent = require(packageJsonFilePath);
+				let isPlugin = fileContent.nativescript;
```

Comment:
Can we make the value of isPlugin a true boolean? Using !! or comparing for null.